src/app/qgshandlebadlayers.cpp

```diff
@@ -270,28 +270,29 @@ void QgsHandleBadLayers::browseClicked()
     const QString provider = mLayerList->item( row, 0 )->data( static_cast< int >( CustomRoles::Provider ) ).toString();
 
     QString fileFilter;
+    QgsProviderMetadata *metadata = QgsProviderRegistry::instance()->providerMetadata( provider );
```

Comment:
There's zero chance here we'd end up with a null pointer (e.g. provider not having a registered metadata counterpart)?

Reply:
I'm not sure -- you'll notice I left the existing logic for all non vector tile layers, but added a nullptr check for the new vector tile handling. I figured there's been no bug reports for the other layer types, so the non-null assumption must be valid for those!